### PR TITLE
Dockerfile: Switch to mage builds, various optimizations

### PR DESCRIPTION
**Pull request by jakejarvis**
Hello all! This is tangentially related to #5056. I'm suggesting the following changes to the current `Dockerfile`:

- Use Alpine in both stages for consistency.
- Switch to `mage install` from `go install` as dictated in [the latest docs](https://github.com/gohugoio/hugo/blob/master/CONTRIBUTING.md).
- Easy switch to building Hugo Extended by directly setting `HUGO_BUILD_TAGS` flag at image build time (simpler than the current `99notag` workaround):

```
docker build --build-arg HUGO_BUILD_TAGS=extended .
```

- Update to Go 1.13 and Alpine 3.10.
- The only possibly breaking change: Moved the `hugo` binary in the final stage from `/hugo` to `/usr/bin/hugo` so one can simply run `hugo` without having to search around for its location.

---

I realize these are big changes and I'm honestly not sure how much the team depends on Docker — it doesn't seem like very much from the outside! — but I don't want to break anyone's workflow so if you'd like me to revert any one of the changes above just let me know. I've been [using](https://github.com/jakejarvis/hugo-custom/blob/master/Dockerfile) almost these exact steps for the past few releases and they've worked great, even with SASS support. 😉 

Thanks as always!

### CODE REVIEW

**Comment by jakejarvis** [comment]
force-pushing to re-trigger Travis — **_completely_** unrelated to this PR but is there some weird non-deterministic race error going on? Not familiar with the tests but it seems odd for this one job to fail out of otherwise successful ones. https://travis-ci.org/gohugoio/hugo/jobs/595643601

**Comment by bep** [comment]
>Not familiar with the tests but it seems odd for this one job to fail out of otherwise successful ones.

https://github.com/gohugoio/hugo/issues/6409 -- we have some global loggers (yes I know) that we naively reset between builds -- I assume this is only a real problem when testing, but we should fix it.

As to the changes itself. I'm not using this Dockerfile myself, but I know other does. I have only one comment on your changes: I would prefer it if we avoided using Mage to build. It doesn't add much usefulness to the mix, and if we would want to use it, I think we need to download a release version and rely on compiling from source.

 @anthonyfok may have something to say.

File: Dockerfile [hunk 1]
```diff
@@ -2,32 +2,44 @@
...
 COPY . /go/src/github.com/gohugoio/hugo/
-ENV GO111MODULE=on
-RUN go get -d .
 
-ARG CGO=0
-ENV CGO_ENABLED=${CGO}
-ENV GOOS=linux
+# gcc/g++ are required to build SASS libraries for extended version
+RUN apk update && \
+    apk add --no-cache gcc g++ musl-dev && \
```
Comment (riyad) [inline Dockerfile]: FYI: `musl-dev` is automatically installed when installing `g++`

**Comment by jufemaiz** [comment]
Just a QQ: https://hub.docker.com/r/gohugoio/hugo seems to be out of date compared with this. The builds are not automated I assume?

https://hub.docker.com/r/gohugoio/hugo/builds

**Comment by peaceiris** [comment]
@jufemaiz 
> That gohugoio account isn't official.
> https://github.com/gohugoio/hugo/issues/4693#issuecomment-518031415

**Comment by jufemaiz** [comment]
@peaceiris eep. Perhaps that should be remedied by the owner?!

**Comment by peaceiris** [comment]
I hope so. Or the Hugo team gets that gohugoio account from the current owner.

**Pull request merged**